plugins/inputs/http_listener_v2/http_listener_v2.go

```diff
@@ -158,16 +166,37 @@ func (h *HTTPListenerV2) Start(acc telegraf.Accumulator) error {
 
 	h.acc = acc
 
-	server := h.createHTTPServer()
+	h.server = h.createHTTPServer()
+
+	go func() {
+		// This goroutine just moves data from the metrics channel to the accumulator
+		// It exists so we can select on writing to the metrics channel and handle cancellation correctly
+		// in the request handler.
+		defer func() {
+			// Handle the case where the accumulator channel is closed while shutting down. We do this to
+			// avoid changing the accumulator API to allow for more graceful handling of this channel.
+			if err := recover(); err != nil {
+				h.Log.Error(err)
+			}
+		}()
```

Comment:
This seems to me like those comments should be shifted each one line higher to actually describe what's underneath them.

Reply:
Good point, done.